test/org/sagebionetworks/bridge/upload/UploadValidationTaskTest.java

```diff
@@ -3,37 +3,191 @@
+            .withSchemaId(SCHEMA_ID)
+            .withSchemaRevision(SCHEMA_REV)
+            .withStudyId(TestConstants.TEST_STUDY_IDENTIFIER)
+            .withUploadDate(UPLOAD_DATE)
+            .withUploadedOn(UPLOADED_ON)
+            .withUploadId(UPLOAD_ID)
+            .withUserSharingScope(ParticipantOption.SharingScope.SPONSORS_AND_PARTNERS)
+            .withUserExternalId(USER_EXTERNAL_ID)
+            .withUserDataGroups(TestConstants.USER_DATA_GROUPS)
+            .withData(BridgeObjectMapper.get().readTree(DATA_TEXT))
+            .withMetadata(BridgeObjectMapper.get().readTree(METADATA_TEXT))
+            .build();
+
+    private HealthDataRecord TEST_RECORD_DUPE = new DynamoHealthDataRecord.Builder()
+            .withCreatedOn(CREATED_ON)
+            .withHealthCode(HEALTH_CODE)
+            .withId(RECORD_ID_2)
+            .withSchemaId(SCHEMA_ID)
+            .withSchemaRevision(SCHEMA_REV)
+            .withStudyId(TestConstants.TEST_STUDY_IDENTIFIER)
+            .withUploadDate(UPLOAD_DATE)
+            .withUploadedOn(UPLOADED_ON)
+            .withUploadId(UPLOAD_ID)
+            .withUserSharingScope(ParticipantOption.SharingScope.SPONSORS_AND_PARTNERS)
+            .withUserExternalId(USER_EXTERNAL_ID)
+            .withUserDataGroups(TestConstants.USER_DATA_GROUPS)
+            .withData(BridgeObjectMapper.get().readTree(DATA_TEXT))
+            .withMetadata(BridgeObjectMapper.get().readTree(METADATA_TEXT))
+            .build();
+
+    private HealthDataRecord TEST_RECORD_DUPE_2 = new DynamoHealthDataRecord.Builder()
+            .withCreatedOn(CREATED_ON)
+            .withHealthCode(HEALTH_CODE)
+            .withId(RECORD_ID_3)
+            .withSchemaId(SCHEMA_ID)
+            .withSchemaRevision(SCHEMA_REV)
+            .withStudyId(TestConstants.TEST_STUDY_IDENTIFIER)
+            .withUploadDate(UPLOAD_DATE)
+            .withUploadedOn(UPLOADED_ON)
+            .withUploadId(UPLOAD_ID)
+            .withUserSharingScope(ParticipantOption.SharingScope.SPONSORS_AND_PARTNERS)
+            .withUserExternalId(USER_EXTERNAL_ID)
+            .withUserDataGroups(TestConstants.USER_DATA_GROUPS)
+            .withData(BridgeObjectMapper.get().readTree(DATA_TEXT))
+            .withMetadata(BridgeObjectMapper.get().readTree(METADATA_TEXT))
+            .build();
+
+    private final List<HealthDataRecord> TEST_RECORD_DUPE_LIST_NORMAL = Arrays.asList(TEST_RECORD, TEST_RECORD_DUPE);
+    private final List<HealthDataRecord> TEST_RECORD_DUPE_LIST_MULTI = Arrays.asList(TEST_RECORD, TEST_RECORD_DUPE, TEST_RECORD_DUPE_2);
+
+    private final List<UploadValidationHandler> handlerList = ImmutableList.of(
+            new MessageHandler("foo was here"), new MessageHandler("bar was here"),
+            new MessageHandler("kilroy was here"), new RecordIdHandler(RECORD_ID));
+
+
+    private HealthDataService healthDataService;
+
+    public UploadValidationTaskTest() throws IOException {
+    }
```

Comment:
Test classes shouldn't have constructors.

Also, we need a unit test for when context.getRecordId() is null.